nvflare/fuel/utils/buffer_list.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class BufferList:
+    """A buffer list that can be treated as a single buffer"""
+
+    def __init__(self, buf_list: list):
+        self.buf_list = buf_list
+
+    def get_size(self):
+
+        if self.buf_list:
+            size = sum(len(buf) for buf in self.buf_list)
+        else:
+            size = 0
```

Comment:
Some enhancement we could do:

If size only changes when "append", should we just have a self.size maintained and increase it in append?
Then we just use that self.size instead of calculating it every time?

Reply:
get_size() is only used once right now. This is not needed right now.